src/pq_flash_index.cpp

```diff
@@ -1393,7 +1374,9 @@ void PQFlashIndex<T, LabelT>::cached_beam_search(const T *query1, const uint64_t
                     if (!use_filter && _dummy_pts.find(id) != _dummy_pts.end())
                         continue;
 
-                    if (use_filter && !point_has_label(id, filter_num) && !point_has_label(id, _universal_filter_num))
+                    if (use_filter && !(point_has_label(id, filter_label)) &&
+                        (!_use_universal_label ||
+                         (_use_universal_label && !point_has_label(id, _universal_filter_label))))
```

Comment:
(!_use_universal_label || !point_has_label(id, _universal_filter_label)) should be good enough?

Reply:
sure. will remove.